proxy/src/telemetry/metrics/histogram.rs

```diff
@@ -77,23 +80,46 @@ impl<V: Into<u64>> Histogram<V> {
             .expect("all values must fit into a bucket");
 
         self.buckets[idx].incr();
-        self.sum += Wrapping(value);
+        self.sum += value;
     }
+}
+
+impl<V: Into<u64>> FmtMetric for Histogram<V> {
+    fn fmt_metric<N: Display>(&self, f: &mut fmt::Formatter, name: N) -> fmt::Result {
+        let mut total = Counter::default();
+        for (le, count) in self.bounds.0.iter().zip(self.buckets.iter()) {
+            total += *count;
+            total.fmt_metric_labeled(f, Key(&name, "bucket"), format!("le=\"{}\"", le))?;
```

Comment:
In my original implementation, I tried to avoid using `format!` and only use `write!`,  since `format!` allocates a `String`. It would be nice, IMO, to be able to pass something that's formatted lazily here,  like what you've done with `Key`. Could cut down on string allocations significantly.

This is probably a premature optimization, but I just wanted to put it out there.

Reply:
yeah, i thought about this too. I'll take a look at this.